### PR TITLE
main features:

### PR DESCRIPTION
**Pull request by SabiMantock**
main features
- [x] Create a app.rb file that will serve as the console app entry-point.
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person). Create a book.
- Create a rental.
- List all rentals for a given person id.
 - [x] In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
- Present the user with a list of options to perform. Lets users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+  def list_people
+    if @people.empty?
+      puts 'There\'s no one here ATM!'
+    else
+      @people.each do |person|
+        puts "Name: #{person.name} Age: #{person.age} ID: #{person.id}"
+      end
+    end
+  end
```
Comment (AmaduKamara) [inline app.rb:44]: - [ ] When I create a student or teacher, the order it displays does not correlate as expected. In the space of the age, it shows the person's name and in space of the name it shows the grade of the person. Kindly fix this so the name shows like this Name: WebAvenger, Age: 24 as well as for the teacher 👍🏽

Screenshot 2023-03-25 at 6 15 43 PM

Screenshot 2023-03-25 at 6 13 31 PM

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  def list_all_rentals(person_id)
+    rentals = @rentals.select { |rental| rental.person.id == person_id }
+    if rentals.empty?
+      puts 'No rentals found for the given person ID!'
+    else
+      rentals.each do |rental|
+        puts "#{rental.book.title} by #{rental.book.author}, rented on #{rental.date}"
+      end
+    end
+  end
```
Comment (AmaduKamara) [inline app.rb:139]: - [ ] I believe it's due to the order in which the data is displayed, when I try to list rental for a person, the app crashes, please work on this and be sure all works fine before submitting 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @SabiMantock,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ App UI is created nicely 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly arrange the order in which the data is displayed in accordance with the label 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Ismailco — COMMENTED** [review]
Hi @SabiMantock, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has some linter errors, kindly make sure to fix them first before re-submitting your project for revew

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by sumairq — APPROVED** [review]
### Hi @SabiMantock , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**